EAD-Session-1-Assignment/functions.js

```diff
@@ -0,0 +1,373 @@
+/**
+ * Class containing methods to validate the form
+ */
+class Validation
+{
+    Validation(){}
+
+    /**
+     * Method to validate the user details
+     * @param {*} name is the user name
+     * @param {*} email is the user email
+     * @param {*} password is the user password
+     * @param {*} confirmPassword is the confirm password field
+     * @param {*} contact is the user contact number
+     * @param {*} gender is the user gender
+     */
+    static validateUserInformation(name , email , password , confirmPassword , contact , gender)
+    {
+        // Validating Contact Number
+        var count=0;
+		var filterContact = new RegExp('([a-zA-Z!@#$%^&*()_+])');
+		if(!(contact.length <= 10 && contact.length >= 8) ) {
+			this.errorDescription('required_e_contact',"Insert Number Between 8 to 10");
+			count++;
+		}else if(filterContact.test(contact)) {
+			this.errorDescription('required_e_contact',"Numeric values are only allowed");
+		} else {
+			this.removeDescription('required_e_contact',"");
+			count--;
+		}
+
+        //Validating user name
+		var filterName = new RegExp('\\d');
+		if(name.length <= 2) {
+			this.errorDescription('required_e_name',"Insert name greater then 2 word");
+			count++;
+		} else if(filterName.test(name)) {
+			this.errorDescription('required_e_name',"Numeric value not allowed");
+			count++;
+		} else{
+			this.removeDescription('required_e_name',"");
+			count--;
+		}
+
+        //Validating email
+		var filterEmail = new RegExp("^([a-zA-Z0-9_\.\-])+\@(([a-zA-Z0-9\-])+\.)+([a-zA-Z0-9]{2,4})+$");
+  		if (!filterEmail.test(email)) {
+			this.errorDescription('required_e_email' ,"Insert a Valid Email");
+			count++;
+		} else {
+			this.removeDescription('required_e_email',"");
+			count--;
+		}
+
+        //Validating Password
+		var filterPassword = new RegExp("^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#\$%\^&\*])(?=.{8,})");
+		if(password.length <= 8){
+			this.errorDescription('required_e_pass' ,"Atleast 8 char required");
+			count++;
+		} else if (!filterPassword.test(password)) {
```

Comment:
Change message like password should have character, numeric values, special character, Upper case letter etc.

Reply:
Done